docs/source/en/migration.md

```diff
@@ -1 +1,300 @@
+
+```js
+const { news, user } = await app.toPromise(ctx.service.biz.list(topic, uid));
+```
+
+#### Others
+
+- thunks (functions)
+- generators (delegation)
+- generator functions (delegation)
+
+Use `async function` to replace the above functions, or use [app.toAsyncFunction] to wrap them if async function is not the case.
+
+**Note**
+- [toAsyncFunction][app.toAsyncFunction] and [toPromise][app.toPromise] are wrappers of [co], thus it may cause performance lost and error stack problems. So we're recommending developers to use all-chain upgrade.
+- [toAsyncFunction][app.toAsyncFunction] doesn't have lost when invokes async function.
+
+@sindresorhus has wrote a lot [promise-based helpers](https://github.com/sindresorhus/promise-fun), use them together with async function could make source code more readable.
+
+## Plugin update
+`App developers` just need to update the upgraded plugins by `plugin developers`, or use `egg-bin autod` command we've prepared to quickly update.
+
+The following content is for `plugin developers`, it shows how to update the plugins:
+
+### Update precautions
+
+- Finish the upgrade items above.
+  - Replace all `generator function` to `async function`.
+  - Upgrade middlewares.
+- Interfaces compabilities (optional), see following.
+- Release a major version.
+
+### Interfaces compabilities
+In some cases, the interface provided by `Plugin developers` supports both generator and async, normally it's wrapped by co.
+
+- In 2.x, we suggest to change to `async-first` in order to get a better performance and clearer error stacks.
+- If necessary, please use [toAsyncFunction][app.toAsyncFunction] and [toPromise][app.toPromise] for compatibilities.
+
+Like [egg-schedule] plugin, it supports generator or async to define the tasks in application level.
+
+```js
+// {app_root}/app/schedule/cleandb.js
+exports.task = function* (ctx) {
+  yield ctx.service.db.clean();
+};
+
+// {app_root}/app/schedule/log.js
+exports.task = async function splitLog(ctx) {
+  await ctx.service.log.split();
+};
+```
+
+`Plugin developers` could simply wrap the following raw function:
+
+```js
+// https://github.com/eggjs/egg-schedule/blob/80252ef/lib/load_schedule.js#L38
+task = app.toAsyncFunction(schedule.task);
+```
+
+### Rules of Plugin release
+
+- **Major version releasement**
+  - All the APIs are promise based, and there's no `async` in source code. e.g. [egg-view-nunjucks]
+- Modify `package.json`
+  - Change `egg` in `devDependencies` to `^2.0.0`.
+  - Change `engines.node` to `>=8.0.0`.
+  - Change `ci.version` to `8, 9`, reinstall dependencies to generate new travis config files.
+- Update examples in `README.md` with async function.
+- Write instructions for upgrade.
+- (optional) Change `test/fixtures` to async function, recommend to create a PR for code preview. 
+
+In case the previous versions still requires maintenance:
```

Comment:
`In case the previous versions still requires maintenance` -> `The previous versions of egg are still under maintenance`